yt/utilities/file_handler.py

```diff
@@ -99,7 +101,12 @@ def valid_netcdf_classic_signature(filename):
 
 
 def warn_netcdf(fn):
-    needs_netcdf = valid_netcdf_classic_signature(fn)
+    # There are a few variants of the netCDF format.
+    classic = valid_netcdf_classic_signature(fn)
+    # NetCDF-4 Classic files are HDF5 files constrained to the Classic
+    # data model used by netCDF-3.
+    netcdf4_classic = valid_hdf5_signature(fn) and fn.endswith('.nc')
```

Comment:
Aww!  My colleague @bstdenis (who, unlike me, has extensive experience working with climate data) tells me that the `.nc` file extension is used 99% of the time, but recently he stumbled across `.nc4`...